Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -1325,42 +1325,80 @@ extension Tensor.IndexPath {
 //===------------------------------------------------------------------------------------------===//
 
 extension Tensor {
-  /// Returns `true` if the given axis is in the range `[-rank, rank)`.
+  /// Returns `true` iff `k` denotes an axis of `self`.
   @usableFromInline
-  internal func isAxisInRange<T: BinaryInteger>(_ axis: T) -> Bool {
-    let axis = Int(axis)
+  internal func isValid<T: BinaryInteger>(axis k: T) -> Bool {
+    let axis = Int(k)
     return axis >= -rank && axis < rank
   }
 
-  /// Returns `true` if the given scalar tensor is in the range `[-rank, rank)`.
+  /// Returns `true` iff each element of `dims` denotes an axis of `self`.
   @usableFromInline
-  internal func isAxisInRange(
-    _ axis: Tensor<Int32>,
+  internal func areValid<T: BinaryInteger>(axes dims: [T]) -> Bool {
+    return dims.allSatisfy { isValid(axis: $0) }
+  }
+
+  /// Returns `true` iff each element of `dims` denotes an axis of `self`.
+  ///
+  /// - Precondition: `dims` is rank 0 or rank 1.
+  @usableFromInline
+  internal func areValid(
+    axes dims: Tensor<Int32>,
     file: StaticString = #file,
     line: UInt = #line
   ) -> Bool {
-    precondition(axis.rank == 0, "Axis must have rank 0.", file: file, line: line)
-    return areAxesInRange(axis.scalars)
+    precondition(
+      dims.rank < 2,
+      "Axes must have rank 0 or rank 1; axes has rank \(dims.rank) with values \(dims.scalars).",
+      file: file,
+      line: line)
+    return areValid(axes: dims.scalars)
+  }
+
+  /// Checks that each element of `dims` denotes an axis of `self`, and stops the program with a
+  /// diagnostic otherwise.
```

Comment:
Just so you know I'm not being arbitrary and thoughtless in my suggestions, there are reasons I thought my suggested phrasing with the semicolon was better, FWIW.  I certainly respect your perogative to make a different choice so I'm not going to get into it unless you really want to discuss it.